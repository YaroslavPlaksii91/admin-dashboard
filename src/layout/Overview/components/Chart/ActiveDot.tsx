import { FC } from 'react';

import { ActiveDotProps } from './types';

export const ActiveDot: FC<ActiveDotProps> = ({ cx, cy }) => {
  return (
    <g>
      <circle
        opacity="0.16"
        cx={cx}
        cy={cy}
        r="13"
        stroke="#3751FF"
        strokeWidth="2"
        fill="none"
      />

      <circle
        cx={cx}
        cy={cy}
        r="4"
        fill="#fff"
        strokeWidth="4px"
        stroke="#3751FF"
        width="14px"
        height="14px"
        filter="drop-shadow(0px 1px 2px rgba(55, 81, 255, 0.20))"
        className="recharts-dot"
      />
    </g>
  );
};
