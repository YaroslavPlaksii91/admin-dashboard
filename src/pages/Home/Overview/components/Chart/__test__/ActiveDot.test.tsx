import { render } from '@testing-library/react';

import { ActiveDot } from '../ActiveDot';

describe('ActiveDot component', () => {
  const cx = 50;
  const cy = 50;

  it('renders ActiveDot correctly', () => {
    const { asFragment } = render(<ActiveDot cx={cx} cy={cy} />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('should render with correct props', () => {
    const { container } = render(<ActiveDot cx={cx} cy={cy} />);

    const circleElements = container.querySelectorAll('circle');
    expect(circleElements).toHaveLength(2);

    const outerCircle = circleElements[0];
    expect(outerCircle.getAttribute('cx')).toBe(cx.toString());
    expect(outerCircle.getAttribute('cy')).toBe(cy.toString());
    expect(outerCircle.getAttribute('r')).toBe('13');
    expect(outerCircle.getAttribute('stroke')).toBe('#3751FF');
    expect(outerCircle.getAttribute('stroke-width')).toBe('2');
    expect(outerCircle.getAttribute('fill')).toBe('none');

    const innerCircle = circleElements[1];
    expect(innerCircle.getAttribute('cx')).toBe(cx.toString());
    expect(innerCircle.getAttribute('cy')).toBe(cy.toString());
    expect(innerCircle.getAttribute('r')).toBe('4');
    expect(innerCircle.getAttribute('fill')).toBe('#fff');
    expect(innerCircle.getAttribute('stroke')).toBe('#3751FF');
    expect(innerCircle.getAttribute('stroke-width')).toBe('4px');
    expect(innerCircle.getAttribute('filter')).toBe(
      'drop-shadow(0px 1px 2px rgba(55, 81, 255, 0.20))',
    );
    expect(innerCircle.getAttribute('class')).toBe('recharts-dot');
  });
});
