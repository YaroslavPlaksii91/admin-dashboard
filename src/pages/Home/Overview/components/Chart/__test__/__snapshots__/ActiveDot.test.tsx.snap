// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActiveDot component renders ActiveDot correctly 1`] = `
<DocumentFragment>
  <g>
    <circle
      cx="50"
      cy="50"
      fill="none"
      opacity="0.16"
      r="13"
      stroke="#3751FF"
      stroke-width="2"
    />
    <circle
      class="recharts-dot"
      cx="50"
      cy="50"
      fill="#fff"
      filter="drop-shadow(0px 1px 2px rgba(55, 81, 255, 0.20))"
      height="14px"
      r="4"
      stroke="#3751FF"
      stroke-width="4px"
      width="14px"
    />
  </g>
</DocumentFragment>
`;
